### ISSUE TITLE
Feat: user properties evolution with permissions

### ISSUE THREAD

**Issue by buxx** [tracim/tracim#4004]
## Feature description and goals

User properties developed in #2935 are currently all visible by admin, owner and other users. This feature permit specifying read/write permissions. Internal documentation is available [here](https://algoo.tracim.fr/ui/workspaces/49/contents/html-document/16718?folder_open=9701,13842).

## Prerequisites

* #2935 is done

## Implementation

- [ ] JSON Schema will be filled with `tracim.permissions` object for each field. This object could contain wollowing keys:
  - "owner" key with value "READ" or "READ/WRITE".
  - "friend" key with value "READ" (:warning: : we don't know what is a friend yet !)
  - "owner" key with value "READ"
  
  Admins always have READ and WRITE permissions
  If permission is not present in `tracim.permissions` object, there is **no** permission

- [ ]  `PUT /api/users/{user_id:\d+}/properties/{property_id}/` check permission of connected user
- [ ] `GET /api/users/{user_id:\d+}/properties/` fill returned object according to connected user  permissions

## Pending questions

- What is a "friend" ?
- Must write a frontend issue to implement (use permissions properties to know what property permit editing)

**Comment by inkhey** [comment]
Does this mean that endpoint will return different results depending on the original user ?
Most of our endpoint try to avoid this.

**Comment by grignards** [comment]
> Does this mean that endpoint will return different results depending on the original user ?
> Most of our endpoint try to avoid this.

Yes, the alternative here would have been individual endpoints for each property, we didn't choose it to ease the usage of the properties API for the user profile use-case.

**Comment by inkhey** [comment]
about "friend", is there any reason to have strict distinction between "friendship" and "following" mechanism ?

**Comment by grignards** [comment]
> about "friend", is there any reason to have strict distinction between "friendship" and "following" mechanism ?

I guess they are related but am not sure they are the same : AFAIK following a user does not require him to accept the follower, so using the follower concept to restrict visible profile entries seems strange to me.

**Comment by inkhey** [comment]
> I guess they are related but am not sure they are the same : AFAIK following a user does not require him to accept the follower, so using the follower concept to restrict visible profile entries seems strange to me.

I'm more thinking it the other way, following someone means accept him to be "our friend". I think about it because it remind me the way presence work in xmpp.

**Comment by grignards** [comment]
I saw that json schema has a [concept of `readOnly`]() property, so the schema we return could be tuned by the permissions without having to expose them in the API.
